elf_binary.cc

```diff
@@ -181,7 +181,15 @@ std::pair<std::string, std::string> ELFBinary::GetVersion(int index) {
                     if (vna->vna_other == versym_[index]) {
                         LOG(INFO) << "Find Elf_Vernaux corresponds to " << versym_[index] << SOLD_LOG_KEY(strtab_ + vn->vn_file)
                                   << SOLD_LOG_KEY(strtab_ + vna->vna_name);
-                        return std::make_pair(std::string(strtab_ + vn->vn_file), std::string(strtab_ + vna->vna_name));
+
+                        std::string filename = std::string(strtab_ + vn->vn_file);
+                        auto found = filename_to_soname.find(filename);
+                        if (found != filename_to_soname.end()) {
+                            return std::make_pair(found->second, std::string(strtab_ + vna->vna_name));
+                        } else {
+                            LOG(WARNING) << "There is no entry for " << filename << " in filename_to_soname. Use filename as soname. ";
```

Comment:
Can this code path actually happen? Shouldn't we always have an entry for all filenames?

Reply:
When a shared object (e.g., libc.so) is supposed not to link by ShouldLink, 
I didn't push its name to  Sold::filename_to_soname_. To overcome this bug, we needed this code path.

But I fixed this bug with https://github.com/shinh/sold/pull/37/commits/527fbd451730916b00de9b6b96ff5c17df5157ca, So I deleted the code path in https://github.com/shinh/sold/pull/37/commits/e99016fa640c84036bee0f8b4904805c1008fc6a.

(Although I said "some shared objects don't have a soname" at today's meeting, it's only my mistake.)